update documentation to include kuberentes initContainers

mickfeech opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

Updating documentation to include details around using initContainers instead up creating a custom image

##### ISSUE TYPE
Docs Pull Request

##### COMPONENT NAME
Docs

##### AWX VERSION
```
make: Nothing to be done for `version'.
```

mickfeech committed:
update documentation to include kuberentes initContainers

ryanpetrello reviewed (approved):
@mickfeech thanks for putting this together, I like it :)

matburt commented on docs/custom_virtualenvs.md:
```diff
@@ -77,6 +77,48 @@ index d69e2c9..a08bae5 100644
+
+    initContainers:
+        - image: 'centos:7'
+        - name: init-my-custom-venv
```
The extra dashes here are going to cause problems, I think it should look like:

```
initContainers:
   - image: ...
     name: ....
     command:
        - sh
        - ....
```

mickfeech replied:
Correct.  Updated.

matburt commented on docs/custom_virtualenvs.md:
```diff
@@ -77,6 +77,48 @@ index d69e2c9..a08bae5 100644
+            - mountPath: /var/lib/awx/venv/my-custom-venv
+              name: custom-venv
+
+Fiinally in the awx-celery container stanza add the shared volume as a mount.
```
This is probably going to need to appear in the `awx-web` container also in order for it to appear as a selectable venv from the web service for `/api/v2/config`

mickfeech replied:
You're right.  Just updated.

matburt commented:
This is a really excellent doc, and is a great alternative to a full rebuild of the image. I'd like to promote this to the general user documentation also.

ryanpetrello commented:
@mickfeech would you mind rebasing against `devel` so we can merge this?

matburt commented:
I've removed the `mergeit` label because I had some feedback above that needs to be addressed and it looks it might still need a rebase.  If we can take care of these I'll reapply the mergeit label.

mickfeech committed:
update documentation to include kuberentes initContainers

Update documentation to include Kubernetes initContainers in custom virtualenvs

mickfeech committed:
update documentation to include kuberentes initContainers

Update documentation to include Kubernetes initContainers in custom virtualenvs

sharkcus commented:
Attempted with version 3.0.1 and I was required to mount the volume in the awx-web container in addition to the awx-celery container before it would recognize the venv in the api. This did not seem to be required in a 1.0.8 instance. Can someone else confirm?

matburt commented:
@sharkcus it does say that in the documentation provided in this PR

> Fiinally in the awx-celery and awx-web containers stanza add the shared volume as a mount.

matburt commented on docs/custom_virtualenvs.md:
```diff
@@ -77,6 +77,48 @@ index d69e2c9..a08bae5 100644
+            - mountPath: /var/lib/awx/venv/my-custom-venv
+              name: custom-venv
+
+Fiinally in the awx-celery and awx-web containers stanza add the shared volume as a mount.
```
Fiinally -> Finally @mickfeech if you could fix that.

mickfeech committed:
Merge branch 'devel' of https://github.com/mickfeech/awx into devel

mickfeech committed:
Fix misspelled word
